protocol/packages/dex/src/impl_/out_remote.rs

```diff
@@ -670,12 +670,12 @@ mod impl_contract {
             ForwardToInnerContinueMsg,
         >
     where
-        OpenIca: Contract<StateResponse = SwapTask::StateResponse>,
-        SwapTask: SwapTaskT
-            + ContractInSwap<TransferOutState, <SwapTask as SwapTaskT>::StateResponse>
-            + ContractInSwap<SwapState, <SwapTask as SwapTaskT>::StateResponse>,
+        OpenIca: Contract,
+        SwapTask: SwapTaskT<StateResponse = OpenIca::StateResponse>
+            + ContractInSwap<TransferOutState, StateResponse = OpenIca::StateResponse>
+            + ContractInSwap<SwapState, StateResponse = OpenIca::StateResponse>,
     {
-        type StateResponse = SwapTask::StateResponse;
+        type StateResponse = OpenIca::StateResponse;
```

Comment:
why does not apply `X as Y` here?

Reply:
Because I used `OpenIca` instead of `SwapTask` to avoid it. The reason this is equivalent is that `SwapTask` is bound to have the same `StateResponse`.

Reply:
my question was not about SwapTask -> OpenIca, but of not applying the disambiguation technique we discussed above, i.e. `<SwapTask as SwapTaskT>::StateResponse`